assets/js/done-task.js

```diff
@@ -0,0 +1,28 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/assets/js/constants.js";
+import createButton from "/assets/js/factory/createButton.js";
+
+function doneTaskEvent(done, edit, inputElement, tools, allTasks, id) {
+  done.addEventListener(CLICK_EVENT, () => {
+    inputElement.classList.add("done");
+    allTasks[id].done = true;
+    done.classList.add(HIDE_CLASS);
+    edit.classList.add(HIDE_CLASS);
+
+    const currentDate = new Date();
+    const diffTime = currentDate.getTime() - allTasks[id].date.getTime();
+    const diffHours = diffTime / (1000 * 60 * 60);
+    const diffDays = Math.ceil(diffHours / 24);
+
+    const completeTimeButtonText =
+      diffDays === 1 ? `Completed in 1 day` : `Completed in ${diffDays} days`;
+    let completeTime = createButton(
+      "complete-time",
+      completeTimeButtonText,
+      "Time to complete the task"
+    );
+
+    allTasks[id].completeTime = diffDays;
```

Comment:
introduce a variable to the task being modified and reference it later instead of accessing the task from the array every time.

```suggestion
    const task = allTasks[id];
    task.done = true;
    ...

    const diffTime = currentDate.getTime() - task.date.getTime();
    ...

    task.completeTime = diffDays;
```